base/reduce.jl

```diff
@@ -1379,3 +1328,65 @@ function _simple_count(::typeof(identity), x::Array{Bool}, init::T=0) where {T}
     end
     return n
 end
+
+function _mapped_eltype(f, A)
+    elT = eltype(A) # try this
+    if elT === Any
+        elT = @default_eltype(A) # try again
+    end
```

Comment:
Does the branch have any advantage over this?
```suggestion
    elT = @default_eltype A
```